src/database-layer/db.ts

```diff
@@ -237,7 +248,9 @@ export abstract class Tx {
 	}
 	public rollback(): Promise<void> {
 		const promise = this._rollback().finally(() => {
-			this.listeners.rollback.forEach(tryFn);
+			this.listeners.rollback.forEach(asyncTryFn);
```

Comment:
If I read this correctly, errors produced by the rollback or end listeners will not be handled, will they (because here we create new promises that nobody subscribes to)? Is it by design?

Reply:
Yes it's by design, we want to fire/forget the errors and it's on the subscriber to handle any errors they have